api/v2/checluster_types.go

```diff
@@ -0,0 +1,639 @@
+	// PVC settings for PostgreSQL database.
+	// +optional
+	Pvc PVC `json:"pvc,omitempty"`
+}
+
+// Che server metrics configuration
+type ServerMetrics struct {
+	// Enables `metrics` the Che server endpoint.
+	// +kubebuilder:default:=true
+	// +optional
+	Enable bool `json:"enable"`
+}
+
+// Configuration settings for installation and configuration of the Kubernetes Image Puller
+// See https://github.com/che-incubator/kubernetes-image-puller-operator
+// +k8s:openapi-gen=true
+type ImagePuller struct {
+	// Install and configure the Community Supported Kubernetes Image Puller Operator. When set to `true` and no spec is provided,
+	// it will create a default KubernetesImagePuller object to be managed by the Operator.
+	// When set to `false`, the KubernetesImagePuller object will be deleted, and the Operator will be uninstalled,
+	// regardless of whether a spec is provided.
+	// If the `spec.images` field is empty, a set of recommended workspace-related images will be automatically detected and
+	// pre-pulled after installation.
+	// Note that while this Operator and its behavior is community-supported, its payload may be commercially-supported
+	// for pulling commercially-supported images.
+	Enable bool `json:"enable"`
+	// A KubernetesImagePullerSpec to configure the image puller in the CheCluster
+	// +optional
+	Spec imagepullerv1alpha1.KubernetesImagePullerSpec `json:"spec"`
+}
+
+// Settings for installation and configuration of the DevWorkspace operator
+// See https://github.com/devfile/devworkspace-operator
+// +k8s:openapi-gen=true
+type DevWorkspace struct {
+	// Deployment override options.
+	// +optional
+	Deployment Deployment `json:"deployment,omitempty"`
+	// Maximum number of the running workspaces per user.
+	// +optional
+	RunningLimit string `json:"runningLimit,omitempty"`
+}
+
+type DefaultNamespace struct {
+	// if users namespaces are not pre-created this field defines the Kubernetes namespace created when a user starts his first workspace.
+	// It's possible to use `<username>`, `<userid>` and `<workspaceid>` placeholders, such as che-workspace-<username>.
+	// +optional
+	Template string `json:"template,omitempty"`
+}
+
+type DashboardHeaderMessage struct {
+	// Instructs dashboard to show the message.
+	// +optional
+	Show bool `json:"show,omitempty"`
+	// Warning message that will be displayed on the User Dashboard
+	// +optional
+	Text string `json:"warning,text"`
+}
+
+type TrustedCerts struct {
+	// The ConfigMap containing certificates to propagate to the Che components and to provide particular configuration for Git.
+	// Note, this ConfigMap must have `app.kubernetes.io/part-of=che.eclipse.org` label.
+	// +optional
+	GitTrustedCertsConfigMapName string `json:"gitTrustedCertsConfigMapName,omitempty"`
+}
+
+// Configuration settings related to the workspaces persistent storage.
+type WorkspaceStorage struct {
+	// PVC settings.
+	// +optional
+	Pvc PVC `json:"pvc,omitempty"`
```

Comment:
If this is for the `claim-che-workspace` PVC in each user's namespace, it's unclear how easy this will be to support. DWO creates PVCs, but will use a `claim-che-workspace` PVC if it is present, so Che could still do this, but I don't know if the functionality exists currently.